libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1782,6 +1786,7 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 				solAssert(functionType.hasDeclaration());
 				solAssert(
 					functionType.kind() == FunctionType::Kind::Error ||
+					functionType.kind() == FunctionType::Kind::Event ||
```

Comment:
I think this needs to be changed. This branch treats `error` and `event` similarly. events have to be treated differently because their signature is 32 bytes.

Reply:
Also to add, there should be an assert that the event is not anonymous.